syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.routeguide";
option java_outer_classname = "AddressServiceProto";
option objc_class_prefix = "AGRC";

package addressparser;

service AddressService {
  rpc Parse(InputAddress) returns (Address) {}
}

message Address {
  string address_number = 1;
  string address_number_suffix = 2;
  string prefix_direction = 3;
  string street_name = 4;
  string street_direction = 5;
  string street_type = 6;
  string unit_type = 7;
  string unit_id = 8;
  string city = 9;
  string zip_code = 10;
  string po_box = 11;
  string normalized = 12;
}

message InputAddress {
  string address = 1;
}
